common/reconciliation/fetcher/timer_test.go

```diff
@@ -0,0 +1,186 @@
+						MinTimestamp:  minTimestamp,
+						MaxTimestamp:  maxTimestamp,
+						BatchSize:     pageSize,
+						NextPageToken: nil,
+					}).
+					Return(&persistence.GetTimerIndexTasksResponse{
+						Timers:        timerTasks,
+						NextPageToken: nil,
+					}, nil)
+
+				mockRetryer.EXPECT().GetShardID().Return(123)
+
+				return mockRetryer
+			},
+			expectedPage: pagination.Page{
+				Entities: []pagination.Entity{
+					&entity.Timer{
+						ShardID:             123,
+						DomainID:            "testDomainID",
+						WorkflowID:          "testWorkflowID",
+						RunID:               "testRunID",
+						TaskType:            persistence.TaskTypeUserTimer,
+						VisibilityTimestamp: fixedTimestamp,
+					},
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name: "Non-nil Pagination Token Provdied",
+			setupMock: func(ctrl *gomock.Controller) *persistence.MockRetryer {
+				mockRetryer := persistence.NewMockRetryer(ctrl)
+
+				mockRetryer.EXPECT().
+					GetTimerIndexTasks(gomock.Any(), &persistence.GetTimerIndexTasksRequest{
+						MinTimestamp:  minTimestamp,
+						MaxTimestamp:  maxTimestamp,
+						BatchSize:     pageSize,
+						NextPageToken: nonNilToken,
+					}).
+					Return(&persistence.GetTimerIndexTasksResponse{
+						Timers:        nil,
+						NextPageToken: nonNilToken,
+					}, nil)
+
+				return mockRetryer
+			},
+			expectedPage: pagination.Page{
+				Entities:     nil,
+				CurrentToken: nonNilToken,
+				NextToken:    nonNilToken,
+			},
+			expectedError: nil,
+		},
+		{
+			name: "Invalid Timer Causes Error",
+			setupMock: func(ctrl *gomock.Controller) *persistence.MockRetryer {
+				mockRetryer := persistence.NewMockRetryer(ctrl)
+
+				invalidTimer := &persistence.TimerTaskInfo{
+					DomainID:            "", // Invalid as it's empty
+					WorkflowID:          "testWorkflowID",
+					RunID:               "testRunID",
+					VisibilityTimestamp: fixedTimestamp,
+					TaskType:            persistence.TaskTypeUserTimer,
+				}
+
+				mockRetryer.EXPECT().
+					GetTimerIndexTasks(gomock.Any(), &persistence.GetTimerIndexTasksRequest{
+						MinTimestamp:  minTimestamp,
+						MaxTimestamp:  maxTimestamp,
+						BatchSize:     pageSize,
+						NextPageToken: nil,
+					}).
+					Return(&persistence.GetTimerIndexTasksResponse{
+						Timers:        []*persistence.TimerTaskInfo{invalidTimer},
+						NextPageToken: nil,
+					}, nil)
+
+				mockRetryer.EXPECT().GetShardID().Return(123)
+
+				return mockRetryer
+			},
+			expectedPage:  pagination.Page{},            // No page is expected due to error
+			expectedError: fmt.Errorf("empty DomainID"), // Adjusted to match the actual error message
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockRetryer := tc.setupMock(ctrl)
+
+			var token pagination.PageToken
+			if tc.name == "Non-nil Pagination Token Provided" {
+				token = nonNilToken
+			}
```

Comment:
why not define a `token` field on test struct and use it from there? we shouldn't have ifs based on test name. `tc` should provide what we need in the test body